lib/saxy/encoder.ex

```diff
@@ -1,13 +1,43 @@
 defmodule Saxy.Encoder do
   @moduledoc false
 
+  @doc """
+  Encode data into a iodata list (merge between list and binary data).
+
+  Example:
+    iex> Saxy.Encoder.encode_to_iodata({"foo", [], []}, [])
+    [[[60, "foo"], 47, 62]]
+
+    iex> Saxy.Encoder.encode_to_iodata({"foo", [{"id", "10"}], []}, [])
+    [[[60, "foo", 32, "id", 61, 34, "10", 34], 47, 62]]
+  """
+  def encode_to_iodata(root, []) do
```

Comment:
I would prefer `nil` instead of empty list since it would communicate that the XML document has no prolog more clearly.

Reply:
good point! I'll modify it.